src/ImageMask/index.js

```diff
@@ -82,6 +82,26 @@ export const ImageMask = ({
         return
       }
       mmgc.clearClassElements()
+      const classBoxes = regions
+        .filter((r) => r.type === "box")
+        .filter((r) => r.cls)
+      for (const classBox of classBoxes) {
+        if (classBox.x < 0 || classBox.x >= 1) continue
+        if (classBox.y < 0 || classBox.y >= 1) continue
```

Comment:
not the correct behavior, if a box is outside the [0,1] bounds it should still be converted into a polygon, but the polygon should not go outside the [0,1] bounds.

The behavior is more generally incorrect, because the bounding box is not being converted into polygon, you're just inserting class points at each of it's corners. This has the effect of allowing the inside of the bounding box to be a different classification of the bounding box (this is bug visible in the screenshot). Use mmgc.addLineToPolygon and construct a 4 line polygon